src/__tests__/theme.spec.ts

```diff
@@ -0,0 +1,257 @@
+        ).reply(200, createdAvailablePageData);
+        mock.onPut(
+            `${URLS.THEME_BY_ID(context.application_id, context.company_id, context.theme_id)}`
+        ).reply(200, updateThemeData);
+        mock.onPut(
+            `${URLS.AVAILABLE_PAGE(context.application_id, context.company_id, context.theme_id)}`
+        ).reply(200, updateAvailablePageData);
+        mock.onPut(
+            `${URLS.THEME_BY_ID(
+                context.application_id,
+                context.company_id,
+                context.theme_id
+            )}/publish`
+        ).reply(200, publishThemeData);
+        mock.onPut(
+            `${URLS.THEME_BY_ID(
+                context.application_id,
+                context.company_id,
+                context.theme_id
+            )}/unpublish`
+        ).replyOnce(200, unpublishThemeData);
+        mock.onGet(
+            `${URLS.GET_APPLICATION_DETAILS(context.application_id, context.company_id)}`
+        ).reply(200, initAppConfigData);
+        mock.onGet(
+            `${URLS.THEME_BY_ID(context.application_id, context.company_id, initThemeData._id)}`
+        ).reply(200, initThemeData);
+
+        let filePath = path.join(process.cwd(), '/src/__tests__/fixtures/archive.zip');
+        let url = initThemeData.src.link;
+        mock.onGet(initThemeData.src.link).reply(function () {
+            return [200, fs.createReadStream(filePath)];
+        });
+    });
+
+    afterEach(() => {
+        configStore.clear();
+    });
+    afterAll(() => {
+        const filePath = path.join(process.cwd() + '/../');
+        try {
+            rimraf.sync(filePath);
+        } catch (err) {
+            console.error(`Error while deleting ${filePath}.`);
+        }
+    });
+    it('should successfully create new theme', async () => {
+        await login();
+        await program.parseAsync([
+            'ts-node',
+            './src/fdk.ts',
+            'theme',
+            'new',
+            '-t',
+            `${token}`,
+            '-n',
+            'rolex',
+        ]);
+        const currentContext = getActiveContext();
+        expect(configObj.application_id).toMatch(currentContext.application_id);
+    });
+
+    it('should successfully sync theme', async () => {
+        await login();
+        const inquirerMock = mockFunction(inquirer.prompt);
+        inquirerMock.mockResolvedValue({ pullConfig: 'Yes' });
+        await program.parseAsync(['ts-node', './src/fdk.ts', 'theme', 'sync']);
+        const currentContext = getActiveContext();
+        expect(configObj.application_id).toMatch(currentContext.application_id);
+    });
+
+    it('should successfully pull config theme', async () => {
+        await login();
+        await program.parseAsync(['ts-node', './src/fdk.ts', 'theme', 'pull-config']);
+        const currentContext = getActiveContext();
+        expect(configObj.application_id).toMatch(currentContext.application_id);
```

Comment:
Add a case where pull-config is needed and compare settings_data.json is updated or not inside theme folder @anuragpandey1115